test/spec/InitSpec.hs

```diff
@@ -4,90 +4,144 @@ module InitSpec where
-          stderr output `shouldBe` "[garn] Creating a garn.ts file\n"
-          readFile "garn.ts"
-            `shouldReturn` unindent
-              [i|
-                import * as garn from "https://garn.io/ts/v0.0.13/mod.ts";
-                import * as pkgs from "https://garn.io/ts/v0.0.13/nixpkgs.ts";
+      it "can initialize go projects" $ \onTestFailureLog -> do
+        writeFile
+          "go.mod"
+          $ unindent
+            [i|
+              module github.com/garnix-io/some-go-project
+              go 1.20
+            |]
+        writeFile "main.go" $
+          unindent
+            [i|
+              package main
 
-                export const someGoProject = garn.go.mkGoProject({
-                  description: "My go project",
-                  src: ".",
-                  goVersion: "1.20",
-                });
-              |]
+              import "fmt"
 
-        it "can initialize npm projects" $ do
-          writeFile
-            "package.json"
+              func main() {
+                fmt.Println("hello world")
+              }
+            |]
+        output <- runGarn ["init"] "" repoDir Nothing
+        onTestFailureLog output
+        stderr output `shouldBe` "[garn] Creating a garn.ts file\n"
+        readFile "garn.ts"
+          `shouldReturn` unindent
             [i|
-              {
-                "name": "my-project",
-                "scripts": {
-                  "start": "echo starting my project..."
-                }
+              import * as garn from "https://garn.io/ts/v0.0.13/mod.ts";
+              import * as pkgs from "https://garn.io/ts/v0.0.13/nixpkgs.ts";
+
+              export const someGoProject = garn.go.mkGoProject({
+                description: "My go project",
+                src: ".",
+                goVersion: "1.20",
+              });
+            |]
+        let check = do
+              output <- runGarn ["build", "someGoProject"] "" repoDir Nothing
+              onTestFailureLog output
+              (StdoutTrim output) <- cmd "./result/bin/some-go-project"
+              output `shouldBe` "hello world"
+        -- TODO: Currently broken on v0.0.13, we should re-enable this on v0.0.14.
+        -- check
+        rewriteImportsToLocalhost
+        check
+
+      it "can initialize npm projects" $ \onTestFailureLog -> do
+        writeFile
+          "package.json"
+          [i|
+            {
+              "name": "my-project",
+              "scripts": {
+                "start": "echo starting my project..."
               }
+            }
+          |]
+        output <- runGarn ["init"] "" repoDir Nothing
+        onTestFailureLog output
+        stderr output `shouldBe` "[garn] Creating a garn.ts file\n"
+        let check = do
+              output <- runGarn ["run", "myProject.start"] "" repoDir Nothing
+              onTestFailureLog output
+              stdout output `shouldContain` "starting my project..."
+        check
```

Comment:
I think this check will cause the problem of failing anytime we introduce a breaking change to the IPC (which is what I ran into on #336). I'm wondering if we should just remove these non-replaced checks